### PR TITLE
XCM v3 Arthur's branch to Sub0 presentation branch

### PR DESCRIPTION
**Pull request by hbulgarini**
@arturgontijo this is a try to sync your trappist with XCM v3 branch with trappist master branch (at this point `sub0-presentation` should be same as master). 

Could you please take a look at the configuration please?

### CODE REVIEW

**Comment by bernardoaraujor** [comment]
I'll try to find some time to review during the retreat.

**Comment by arturgontijo** [comment]
Hey @hbulgarini, thanks for starting this sync saga! =)

What about the other repos? I think we should update them all, in order to use them in all those `Cargo.toml` files:

https://github.com/paritytech/polkadot/tree/trappist-xcm-v3
https://github.com/paritytech/cumulus/tree/trappist-xcm-v3
https://github.com/paritytech/trappist/tree/xcm-v3 (maybe changing the base branch here to this one?)

File: contracts/rwa/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,38 @@
+[package]
+name = "rwa"
```
Comment (bernardoaraujor) [inline contracts/rwa/Cargo.toml:2]: what does `rwa` mean?

**Review by bernardoaraujor — COMMENTED** [review]
My knowledge isn't enough for me to comment on the details of the deep changes to XCM Config.
I'm just assuming that they're needed to keep up with the latest upstream changes. It compiled fine on my machine, although I got some warnings:
```
warning: use of deprecated type alias `sp_api_hidden_includes_construct_runtime::hidden_include::weights::DispatchClass`: Enum has moved to `frame_support::dispatch`
  --> runtime/src/lib.rs:52:23
   |
52 |         ConstantMultiplier, DispatchClass, Weight,
   |                             ^^^^^^^^^^^^^
   |
   = note: `#[warn(deprecated)]` on by default

warning: use of deprecated type alias `sp_api_hidden_includes_construct_runtime::hidden_include::weights::DispatchClass`: Enum has moved to `frame_support::dispatch`
   --> runtime/src/lib.rs:142:14
    |
142 |         .for_class(DispatchClass::all(), |weights| {
    |                    ^^^^^^^^^^^^^

warning: use of deprecated type alias `sp_api_hidden_includes_construct_runtime::hidden_include::weights::DispatchClass`: Enum has moved to `frame_support::dispatch`
   --> runtime/src/lib.rs:145:14
    |
145 |         .for_class(DispatchClass::Normal, |weights| {
    |                    ^^^^^^^^^^^^^

warning: use of deprecated type alias `sp_api_hidden_includes_construct_runtime::hidden_include::weights::DispatchClass`: Enum has moved to `frame_support::dispatch`
   --> runtime/src/lib.rs:148:14
    |
148 |         .for_class(DispatchClass::Operational, |weights| {
    |                    ^^^^^^^^^^^^^

warning: `trappist-runtime` (lib) generated 4 warnings

```

aside from that, I assume it looks good overall.

I would only point out what @arturgontijo said regarding branches. Since we already have the `xcm-v3` it might make sense to keep just a single branch with the latest XCMv3 updates.

File: contracts/rwa/Cargo.toml (hunk 1 above)
Comment (hbulgarini) [reply contracts/rwa/Cargo.toml:2]: Real World Asset .. anyway that contract should not be here ... i will delete it.

**Comment by hbulgarini** [comment]
> Hey @hbulgarini, thanks for starting this sync saga! =)
> 
> What about the other repos? I think we should update them all, in order to use them in all those `Cargo.toml` files:
> 
> https://github.com/paritytech/polkadot/tree/trappist-xcm-v3 https://github.com/paritytech/cumulus/tree/trappist-xcm-v3 https://github.com/paritytech/trappist/tree/xcm-v3 (maybe changing the base branch here to this one?)

i don't think we should change anything on other repos. This PR is mainly syncing your trappist xcm branch with the latest changes so cumulus and polkadot xcmv3 repos remain the same.

**Comment by hbulgarini** [comment]
> My knowledge isn't enough for me to comment on the details of the deep changes to XCM Config. I'm just assuming that they're needed to keep up with the latest upstream changes. It compiled fine on my machine, although I got some warnings:
> 
> ```
> warning: use of deprecated type alias `sp_api_hidden_includes_construct_runtime::hidden_include::weights::DispatchClass`: Enum has moved to `frame_support::dispatch`
>   --> runtime/src/lib.rs:52:23
>    |
> 52 |         ConstantMultiplier, DispatchClass, Weight,
>    |                             ^^^^^^^^^^^^^
>    |
>    = note: `#[warn(deprecated)]` on by default
> 
> warning: use of deprecated type alias `sp_api_hidden_includes_construct_runtime::hidden_include::weights::DispatchClass`: Enum has moved to `frame_support::dispatch`
>    --> runtime/src/lib.rs:142:14
>     |
> 142 |         .for_class(DispatchClass::all(), |weights| {
>     |                    ^^^^^^^^^^^^^
> 
> warning: use of deprecated type alias `sp_api_hidden_includes_construct_runtime::hidden_include::weights::DispatchClass`: Enum has moved to `frame_support::dispatch`
>    --> runtime/src/lib.rs:145:14
>     |
> 145 |         .for_class(DispatchClass::Normal, |weights| {
>     |                    ^^^^^^^^^^^^^
> 
> warning: use of deprecated type alias `sp_api_hidden_includes_construct_runtime::hidden_include::weights::DispatchClass`: Enum has moved to `frame_support::dispatch`
>    --> runtime/src/lib.rs:148:14
>     |
> 148 |         .for_class(DispatchClass::Operational, |weights| {
>     |                    ^^^^^^^^^^^^^
> 
> warning: `trappist-runtime` (lib) generated 4 warnings
> ```
> 
> aside from that, I assume it looks good overall.
> 
> I would only point out what @arturgontijo said regarding branches. Since we already have the `xcm-v3` it might make sense to keep just a single branch with the latest XCMv3 updates.

where are you getting this error from? i did not ... could you please try to run `cargo clean` and the try to build?

**Comment by arturgontijo** [comment]
> i don't think we should change anything on other repos. This PR is mainly syncing your trappist xcm branch with the latest changes so cumulus and polkadot xcmv3 repos remain the same.

But I'm using my personal repos (polkadot and cumulus) in all `Cargo.toml` here...
I'll update those "official" `trappist-xcm-v3` branches, as they were created exactly for that purpose =)

**Comment by hbulgarini** [comment]
> > i don't think we should change anything on other repos. This PR is mainly syncing your trappist xcm branch with the latest changes so cumulus and polkadot xcmv3 repos remain the same.
> 
> But I'm using my personal repos (polkadot and cumulus) in all `Cargo.toml` here... I'll update those "official" `trappist-xcm-v3` branches, as they were created exactly for that purpose =)

Ok! i think i get your point now, the reason why i'm merging the changes to antoher branch is because i want to lock changes for the sub0 presentation, so nothing gets unexpectedly broken ..

**Comment by bernardoaraujor** [comment]
@hbulgarini did `cargo clean` and still getting the warnings.
I simply cloned the repo on `sub0-presentation` and compiled without any modifications.

**Comment by arturgontijo** [comment]
The `trappist-xcm-v3` branches are up to date now:

https://github.com/paritytech/polkadot/tree/trappist-xcm-v3
https://github.com/paritytech/cumulus/tree/trappist-xcm-v3

And the Trappist one, still need to be sinched with master...will do that later today:
https://github.com/paritytech/trappist/tree/xcm-v3

Heads up that all of them are already using `v0.9.31`.

**Comment by evilrobot-01** [comment]
@hbulgarini @bernardoaraujor fyi - I also got those deprecation warnings when I did the update to 0.9.30, but the changes were copied like-for-like from upstream sources, so assumed it was safe for now, especially due to time constraints. Let me know if you'd like me to look into getting them resolved properly.

**Comment by hbulgarini** [comment]
We will drop the branch `arturgontijo-xcm-v3`, so this PR can be closed without merging.

**Pull request closed without merging**